torchgeo/models/farseg.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Foreground-Aware Relation Network (FarSeg) implementations."""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchvision.models import resnet
+from torchvision.ops import FeaturePyramidNetwork as FPN
+from typing import List, Tuple
+import math
+from collections import OrderedDict
+
+
+class FarSeg(nn.Module):
+    """Foreground-Aware Relation Network (FarSeg).
+
+    'Foreground-Aware Relation Network for Geospatial Object Segmentation
+    in High Spatial Resolution Remote Sensing Imagery', Zheng et al. (2020)
+    If you use this model in your research, please cite the following paper:
+
+    * https://arxiv.org/pdf/2011.09766.pdf
+    @inproceedings{zheng2020foreground,
+      title={Foreground-Aware Relation Network for Geospatial Object Segmentation in High Spatial Resolution Remote Sensing Imagery},
+      author={Zheng, Zhuo and Zhong, Yanfei and Wang, Junjue and Ma, Ailong},
+      booktitle={Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition},
+      pages={4096--4105},
+      year={2020}
+    }
+    """
+
+    def __init__(self,
+                 in_channels: int,
+                 backbone: str,
+                 classes: int,
+                 backbone_pretrained: bool = True
+                 ) -> None:
+        """Initializes the FarSeg.
+
+        Args:
+            in_channels: number of channels per input image
+            backbone: name of ResNet backbone
+            classes: number of output segmentation classes
+            backbone_pretrained: whether to use pretrained weight for backbone
+                (default=True)
```

Comment:
I don't think this really works unless `in_channels=3`. In the long run we're planning on shipping resnets pre-trained on multispectral satellite imagery, but we don't have that ready yet. Until then it might be better just to skip this, or raise an error if someone requests a pre-trained model when `in_channels != 3`.